Assets/Scripts/Utilities/Logger.cs

```diff
@@ -0,0 +1,134 @@
+    {
+        Default
+    }
+
+    static Level minimumLevel = Level.Info;
+
+    public static void LogException(Exception exception)
+    {
+        // Exceptions are special as they support neither formatting nor message
+        if (minimumLevel <= Level.Exception)
+        {
+            Debug.LogException(exception);
+        }
+    }
+
+    public static void LogAssertion(string message)
+    {
+        Log(Level.Assertion, message);
+    }
+
+    public static void LogAssertionFormat(string format, params object[] args)
+    {
+        Log(Level.Assertion, format, args);
+    }
+
+    public static void LogError(string message)
+    {
+        Log(Level.Error, message);
+    }
+
+    public static void LogErrorFormat(string format, params object[] args)
+    {
+        Log(Level.Error, format, args);
+    }
+
+    public static void LogWarning(string message)
+    {
+        Log(Level.Warning, message);
+    }
+
+    public static void LogWarningFormat(string format, params object[] args)
+    {
+        Log(Level.Warning, format, args);
+    }
+
+    public static void LogInfo(string message)
+    {
+        Log(Level.Info, message);
+    }
+
+    public static void LogInfoFormat(string format, params object[] args)
+    {
+        Log(Level.Info, format, args);
+    }
+
+    public static void LogVerbose(string message)
+    {
+        Log(Level.Verbose, message);
+    }
+
+    public static void LogVerboseFormat(string format, params object[] args)
+    {
+        Log(Level.Verbose, format, args);
+    }
+
+    /// <summary>
+    /// Alias for info.
+    /// </summary>
+    public static void Log(string message)
+    {
+        Log(Level.Info, message);
+    }
+
+    /// <summary>
+    /// Alias for info.
+    /// </summary>
+    public static void LogFormat(string message, params object[] args)
+    {
+        Log(Level.Info, message, args);
+    }
+
+    public static void Log(Level level, string message, params object[] formatArgs)
+    {
+        if (level > minimumLevel)
+        {
+            return;
+        }
+
+        if (formatArgs.Length > 0)
+        {
+            message = string.Format(message, formatArgs);
+        }
+
+        switch (level)
+        {
+            case Level.Exception:
+                throw new InvalidOperationException("Exceptions are logged in a special way. You shouldnt see this message");
+            case Level.Assertion:
+                Debug.LogAssertion(message);
+                break;
+            case Level.Error:
+                Debug.LogError(message);
+                break;
+            case Level.Warning:
+                Debug.LogWarning(message);
+                break;
+            case Level.Info:
+                Debug.Log(message);
+                break;
+            case Level.Verbose:
+                Debug.Log(message);
+                break;
+            default:
+                throw new ArgumentException("Unhandled level: " + level, "level");    
```

Comment:
Indeed. I say: try to recover, if you can’t crash early and often ;)

 

From: Alexander Fast [mailto:notifications@github.com] 
Sent: Thursday, August 18, 2016 4:11 PM
To: TeamPorcupine/ProjectPorcupine <ProjectPorcupine@noreply.github.com>
Cc: Oswald Maskens <ozie@live.be>; Comment <comment@noreply.github.com>
Subject: Re: [TeamPorcupine/ProjectPorcupine] Added custom logging class wrapping unity logger. (#185)

 

In Assets/Scripts/Utilities/Logger.cs <https://github.com/TeamPorcupine/ProjectPorcupine/pull/185#discussion_r75379898> :

> +                Debug.LogAssertion(message);
> +                break;
> +            case Level.Error:
> +                Debug.LogError(message);
> +                break;
> +            case Level.Warning:
> +                Debug.LogWarning(message);
> +                break;
> +            case Level.Info:
> +                Debug.Log(message);
> +                break;
> +            case Level.Verbose:
> +                Debug.Log(message);
> +                break;
> +            default:
> +                throw new ArgumentException("Unhandled level: " + level, "level");    

Crashing is not bad. As long as it happens at the developer. It's a mistake you can't miss. A logged error message you can miss.

Crashes at the end user is bad. And for this case, it means that code has reached the end user without ever being run. Which is really bad. And for that case I'm willing to risk a crash.

But to be fair, "to crash or not to crash" is one of the holy wars at work. I say crash early and often, some say crash never.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub <https://github.com/TeamPorcupine/ProjectPorcupine/pull/185/files/4203baa6329c320464d14890fc126b677c83eb3b#r75379898> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AH2O3XDwtFgJ-hhzZPrWe3p7iS1EAvB7ks5qhLxOgaJpZM4Jnycm> .  <https://github.com/notifications/beacon/AH2O3XWPrkXOWqdAduEU3KtJptWcN0M6ks5qhLxOgaJpZM4Jnycm.gif>